api/v1alpha1/instaslice_types.go

```diff
@@ -56,105 +33,161 @@ const (
+	Resources corev1.ResourceRequirements `json:"resources"`
+
+	// podRef is a reference to the Pod requesting the allocation
+	// +kubebuilder:validation:Required
+	PodRef corev1.ObjectReference `json:"podRef"`
+}
+
+type AllocationResult struct {
+	// migPlacement specifies the MIG placement details
+	// +kubebuilder:validation:Required
+	MigPlacement Placement `json:"migPlacement"`
+
+	// gpuUUID represents the UUID of the selected GPU
+	// +kubebuilder:validation:Required
 	GPUUUID string `json:"gpuUUID"`
-	// nodename represents name of the selected node
-	// +required
-	Nodename string `json:"nodename"`
-	// allocationStatus represents status of allocation
-	// +kubebuilder:validation:Enum:=deleted;deleting;ungated;creating;created
-	// +required
-	Allocationstatus AllocationStatus `json:"allocationStatus"`
-	// resourceIdentifier represents uuid used for creating configmap resource
-	// +required
-	Resourceidentifier string `json:"resourceIdentifier"`
-	// namespace represents namespace of user workload
-	// +required
-	Namespace string `json:"namespace"`
-	// podName represents name of the user workload pod
-	// +required
-	PodName string `json:"podName"`
-	// cpu represents amount of cpu requested by user workload
-	// +required
-	Cpu int64 `json:"cpu"`
-	// memory represents amount of memory requested by user workload
-	// +required
-	Memory int64 `json:"memory"`
+
+	// nodename represents the name of the selected node
+	// +kubebuilder:validation:Required
+	Nodename types.NodeName `json:"nodename"`
+
+	// allocationStatus represents the current status of the allocation
+	// +kubebuilder:validation:Enum=deleted;deleting;ungated;creating;created
+	// +kubebuilder:validation:Required
+	AllocationStatus AllocationStatus `json:"allocationStatus"`
+
+	// configMapResourceIdentifier represents the UUID used for creating the ConfigMap resource
+	// +kubebuilder:validation:Required
+	ConfigMapResourceIdentifier types.UID `json:"configMapResourceIdentifier"`
+
+	// conditions provide additional information about the allocation
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+type DiscoveredGPU struct {
+	// gpuUUID represents the UUID of the GPU
+	// +kubebuilder:validation:Required
+	GPUUUID string `json:"gpuUuid"`
+
+	// gpuName represents the name of the GPU
+	// +kubebuilder:validation:Required
+	GPUName string `json:"gpuName"`
+
+	// gpuMemory represents the memory capacity of the GPU
+	// +kubebuilder:validation:Required
+	GPUMemory resource.Quantity `json:"gpuMemory"`
+}
+
+type DiscoveredNodeResources struct {
+	// nodeGPUs represents the discovered GPUs on the node
+	// +kubebuilder:validation:Required
+	NodeGPUs []DiscoveredGPU `json:"nodeGPUs"`
+
+	// migPlacement represents GPU instance, compute instance with placement for a profile
+	// +kubebuilder:validation:Required
+	MigPlacement map[MigSliceProfile]Mig `json:"migPlacement"`
+
+	// nodeResources represents the resource list of the node
```

Comment:
nodeResources is a heuristic; these are resources present on the node at boot time, we should mention that in godocs.

Reply:
Sure, I will update that godoc.

Reply:
"node on boot time"

there is existing work to add memory/cpu to existing nodes without a reboot. Do we have this in mind for NodeResources?